client/sources/common/models.py

```diff
@@ -0,0 +1,60 @@
+"""Models for serialization of tests."""
+
+from client.sources.common import core
+
+class Test(core.Serializable):
+    name = core.String()
+    points = core.Float()
+    partner = core.Int(optional=True)
+
+    def run(self):
```

Comment:
I opt not to raise NotImplementedError because some tests might choose not to use some of these methods, in which case the default behavior should just be to do nothing, not error.